### PR TITLE
Feature/ge 709

### PR DESCRIPTION
**Pull request by AltusBarry**

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+MIGRATE = Migrate(APP, DB)
+
+
+class Domain(DB.Model):
+    domain = DB.relationship("Domain", backref="domain", lazy=True)
+    id = DB.Column(DB.Integer, primary_key=True)
+    parent_id = DB.Column(DB.Integer, DB.ForeignKey("domain.id"), nullable=True)
+    name = DB.Column(DB.VARCHAR(30), unique=True, index=True)
+    description = DB.Column(DB.Text)
+    created_at = DB.Column(DB.DateTime, default=datetime.datetime.utcnow())
```
Comment (cobusc) [inline models.py]: Use `default=datetime.datetime.utcnow` instead of calling the function...for all definitions like this.

File: models.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    "DB_URI",
+    "postgres+psycopg2://core-access-control" \
+    ":core-access-control@localhost:5432/core-access-control"
+)
+DB = SQLAlchemy(APP)
+MIGRATE = Migrate(APP, DB)
+
+
+class Domain(DB.Model):
+    domain = DB.relationship("Domain", backref="domain", lazy=True)
```
Comment (cobusc) [inline models.py]: Remove relationships for now.

File: models.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    def __repr__(self):
+        return "<Domain(%s)>" % self.name
+
+
+class Role(DB.Model):
+    role = DB.relationship("Role", backref="role", lazy=True)
+    id = DB.Column(DB.Integer, primary_key=True)
+    label = DB.Column(DB.VARCHAR(30), unique=True, index=True)
+    description = DB.Column(DB.Text)
+    requires_2fa = DB.Column(DB.Boolean(), default=True)
```
Comment (cobusc) [inline models.py]: Why the `()` after `Boolean()`?

Comment (AltusBarry) [reply models.py]: If I had to be honest. I probably copied the VARCHAR lines, change the method and cleared out the ().
In other words, not intentional.

File: models.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    )
+
+    def __repr__(self):
+        return "<UserDomainRole(%s-%s-%s)>" % (
+            self.user_id, self.domain_id, self.role_id
+        )
+
+
+class Invitation(DB.Model):
+    id = DB.Column(UUID, default=uuid.uuid1(), primary_key=True)
```
Comment (cobusc) [inline models.py]: `uuid.uuid1` (without the parenthesis)

File: models.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    first_name = DB.Column(DB.Text)
+    last_name = DB.Column(DB.Text)
+    email = DB.Column(DB.VARCHAR(100), unique=True, index=True)
+    expires_at = DB.Column(DB.DateTime)
+    invitor_id = DB.Column(UUID, default=uuid.uuid1())
```
Comment (cobusc) [inline models.py]: No default. This needs to be provided.

File: models.py [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    updated_at = DB.Column(DB.DateTime, onupdate=datetime.datetime.utcnow())
+
+    def __repr__(self):
+        return "<Permission(%s)>" % self.name
+
+
+class Resource(DB.Model):
+    resource = DB.relationship("Resource", backref="resource", lazy=True)
+    id = DB.Column(DB.Integer, primary_key=True)
+    urn = DB.Column(DB.VARCHAR(100), unique=True, index=True)
```
Comment (cobusc) [inline models.py]: `unique=True` should imply `index=True`

File: models.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    domain_id = DB.Column(DB.Integer, DB.ForeignKey("domain.id"), index=True)
+    description = DB.Column(DB.Text)
+    client_id = DB.Column(DB.Integer, unique=True, index=True)
+    is_active = DB.Column(DB.Boolean, default=True)
+    created_at = DB.Column(DB.DateTime, default=func.utc_timestamp())
+    updated_at = DB.Column(DB.DateTime, onupdate=func.utc_timestamp())
+
+    def __repr__(self):
+        return "<Site(%s-%s-%s)>" % (
+            self.client_id, self.is_active, self.domain_id
```
Comment (cobusc) [inline models.py]: Use name in `__repr__`, please.

**Comment by cobusc** [comment]
:+1:

**Pull request closed without merging**